### ISSUE TITLE
Error: Failed to install 'LandSCENT' from GitHub: stderr is not a pipe

### ISSUE THREAD

**Issue by Arcadi2020** [ChenWeiyan/LandSCENT#14]
Hi Chen,

I am trying to install LandSCENT via GitHub

My sessioninfo():
R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] swirl_2.4.5      igraph_1.2.6     paxtoolsr_1.22.0 XML_3.99-0.5     rJava_0.9-13    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        pillar_1.4.7      compiler_4.0.3    plyr_1.8.6        R.methodsS3_1.8.1 R.utils_2.10.1   
 [7] bitops_1.0-6      tools_4.0.3       testthat_3.0.0    digest_0.6.27     jsonlite_1.7.1    lifecycle_0.2.0  
[13] tibble_3.0.4      pkgconfig_2.0.3   rlang_0.4.8       cli_2.2.0         rstudioapi_0.13   curl_4.3         
[19] yaml_2.2.1        httr_1.4.2        stringr_1.4.0     vctrs_0.3.5       hms_0.5.3         glue_1.4.2       
[25] R6_2.5.0          fansi_0.4.1       readr_1.4.0       magrittr_2.0.1    ellipsis_0.3.1    assertthat_0.2.1 
[31] utf8_1.1.4        stringi_1.5.3     RCurl_1.98-1.2    crayon_1.3.4      rjson_0.2.20      R.oo_1.24.0    

#### THE ERROR MESSAGE:
> devtools::install_github("ChenWeiyan/LandSCENT")
Downloading GitHub repo ChenWeiyan/LandSCENT@HEAD
✓  checking for file ‘/private/var/folders/px/ncqctk414lx71h5wq13b91lcvpnkxl/T/Rtmp5ifQqs/remotes9ee28480512/ChenWeiyan-LandSCENT-ea35ec9/DESCRIPTION’ ...
─  preparing ‘LandSCENT’:
✓  checking DESCRIPTION meta-information ...
─  installing the package to process help pages
Error: Failed to install 'LandSCENT' from GitHub:
  stderr is not a pipe.

Thank you
Arcadi

**Comment by ChenWeiyan** [comment]
Hi Arcadi,

According to `stderr is not a pipe`, It seems the real error of the installation is not shown. This is something related to your operation system. Probably you need to solve it first, and then the installation error will appear? 

Best,
Weiyan

**Comment by Arcadi2020** [comment]
Thank you for the prompt reply, Weiyan

I can install other packages from GitHub, but not this one.
A colleague of mine, has the same problem.

Kind regards
Arcadi

**Comment by Arcadi2020** [comment]
SOLUTION TO: Failed to install 'LandSCENT' from GitHub: stderr is not a pipe

thanks to the help of a (clever) senior bioinformatician, I managed to solve the problem:

I am running R4.0.3 on macOS 10.15.7.
Apparently the package has been only tested using R3.6 on Linux 3.10.
There are a few breaking changes between R3.6.3 and R4.0, so this kind of problem will become increasingly common.
Although macOS is very similar to Linux, there are some differences in core system libraries.
Sometimes it is necessary to install additional libraries that Linux uses take for granted, but which are not included on Macs by default.
In this case, I needed: https://www.xquartz.org/
After downloading, I restarted the laptop and the installation of the package was successful.

Cheers
Arcadi
